frontend/src/modules/Grid3DIntersection/vtkUtils.ts

```diff
@@ -0,0 +1,235 @@
+            uint8[offset++] = (tmp >> 8) & 0xff;
+            uint8[offset++] = tmp & 0xff;
+            break;
+        case 2:
+            while (!isValidChar(b64Str[charIdx])) {
+                charIdx++;
+            }
+            tmp = REVERSE_LOOKUP[b64Str.charCodeAt(charIdx++)] << 2;
+            while (!isValidChar(b64Str[charIdx])) {
+                charIdx++;
+            }
+            tmp |= REVERSE_LOOKUP[b64Str.charCodeAt(charIdx++)] >> 4;
+            uint8[offset++] = tmp & 0xff;
+            break;
+        case 1:
+            throw new Error('BASE64: remain 1 should not happen');
+        case 0:
+            break;
+        default:
+            break;
+    }
+    console.log("offset", offset)
+
+    return offset;
+}
+
+export function toArrayBuffer(b64Str: any): ArrayBuffer {
+    console.log(b64Str)
+    b64Str = b64Str["bvals"]
+
+    const chunks = extractChunks(b64Str);
+    console.log(chunks)
+    const totalEncodedLength = chunks[chunks.length - 1].end + 1;
+    const padding = (4 - (totalEncodedLength % 4)) % 4; // -length mod 4
+    // Any padding chars in the middle of b64Str is to be interpreted as \x00,
+    // whereas the terminating padding chars are to be interpreted as literal padding.
+    const totalSize = ((totalEncodedLength + padding) * 3) / 4 - padding;
+
+    const arrayBuffer = new ArrayBuffer(totalSize);
+    const view = new Uint8Array(arrayBuffer);
+    let dstOffset = 0;
+    for (let i = 0; i < chunks.length; i++) {
+        dstOffset += writeChunk(b64Str, chunks[i], dstOffset, view);
+        dstOffset += (4 - (chunks[i].count % 4)) % 4;
+    }
+
+    return arrayBuffer;
+}
+
+function encodeTriplet(v1: any, v2: any, v3: any) {
+    const triplet = (v1 << 16) + (v2 << 8) + v3;
+    return (
+        BASE64_CODE[triplet >> 18] +
+        BASE64_CODE[(triplet >> 12) & 0x3f] +
+        BASE64_CODE[(triplet >> 6) & 0x3f] +
+        BASE64_CODE[triplet & 0x3f]
+    );
+}
+
+export function fromArrayBuffer(ab: any) {
+    const uint8 = new Uint8Array(ab);
+    const leftoverLength = ab.byteLength % 3;
+    const maxTripletIndex = ab.byteLength - leftoverLength;
+    const segments = Array(maxTripletIndex / 3);
+    for (let i = 0; i < segments.length; i++) {
+        const bufOffset = i * 3;
+        segments[i] = encodeTriplet(
+            uint8[bufOffset],
+            uint8[bufOffset + 1],
+            uint8[bufOffset + 2]
+        );
+    }
+    if (leftoverLength > 0) {
+        const segment = encodeTriplet(
+            uint8[maxTripletIndex],
+            uint8[maxTripletIndex + 1] || 0,
+            uint8[maxTripletIndex + 2] || 0
+        );
+        if (leftoverLength === 1) {
+            segments.push(`${segment.substr(0, 2)}==`);
+        } else if (leftoverLength === 2) {
+            segments.push(`${segment.substr(0, 3)}=`);
+        }
+    }
+    return segments.join('');
+}
+
+export default {
+    toArrayBuffer,
+    fromArrayBuffer,
+};
```

Comment:
We already have a `modules/shared-types` folder. Move this to a `modules/shared-utils` folder? Maybe not its final destination, but easier to move shared code location than later detecting duplicate code. 🙂